var/spack/repos/builtin/packages/neurodamus-model/package.py

```diff
@@ -112,50 +101,39 @@ def install(self, spec, prefix):
-            install(prefix)
+        self._install_binaries()
 
         # Install mods/hocs, and a builder script
         self._install_src(prefix)
         shutil.move(_BUILD_NEURODAMUS_FNAME, prefix.bin)
 
         # Create mods links in share
-        force_symlink(spec['neurodamus-core'].prefix.mod, prefix.share.mod_neurodamus)
+        force_symlink(spec['neurodamus-core'].prefix.lib.mod, prefix.share.mod_neurodamus)
         force_symlink(prefix.lib.mod, prefix.share.mod_full)
 
-        if spec.satisfies('+python'):
-            py_src = spec['neurodamus-core'].prefix.python
-            assert os.path.isdir(py_src)
-            # Link required paths, create a new lib link (to our lib)
-            py_dst = prefix.lib.python
-            mkdirp(py_dst)
-            force_symlink('../../lib', py_dst.lib)
-            for name in ('neurodamus', 'init.py', '_debug.py'):
-                force_symlink(py_src.join(name), py_dst.join(name))
-
-        filter_file(r'UNKNOWN_NEURODAMUS_MODEL', r'%s' % spec.name, prefix.lib.hoc.join('defvar.hoc'))
-        filter_file(r'UNKNOWN_NEURODAMUS_VERSION', r'%s' % spec.version, prefix.lib.hoc.join('defvar.hoc'))
+        filter_file(r'UNKNOWN_NEURODAMUS_MODEL', r'%s' % spec.name,
+                    prefix.lib.hoc.join('defvar.hoc'))
+        filter_file(r'UNKNOWN_NEURODAMUS_VERSION', r'%s' % spec.version,
+                    prefix.lib.hoc.join('defvar.hoc'))
 
         try:
             commit_hash = self.fetcher[0].get_commit()
-            filter_file(r'UNKNOWN_NEURODAMUS_HASH', r'\'%s\'' % commit_hash, prefix.lib.hoc.join('defvar.hoc'))
         except Exception as e:
             tty.warn(e)
+        else:
+            filter_file(r'UNKNOWN_NEURODAMUS_HASH', r"'%s'" % commit_hash[:8],
+                        prefix.lib.hoc.join('defvar.hoc'))
 
     def setup_environment(self, spack_env, run_env):
-        SimModel.setup_environment(self, spack_env, run_env)
-        run_env.set('HOC_LIBRARY_PATH', self.prefix.lib.hoc)
-
-        if self.spec.satisfies("+python"):
-            pylib = self.prefix.lib.python
-            for m in spack_env.env_modifications:
-                if m.name == 'PYTHONPATH':
-                    run_env.prepend_path('PYTHONPATH', m.value)
-            run_env.prepend_path('PYTHONPATH', pylib)
-            run_env.set('NEURODAMUS_PYTHON', pylib)
+        self._setup_environment_common(spack_env, run_env)
+        for libnrnmech_name in find(self.prefix.lib, 'libnrnmech*.so', recursive=False):
+            # We have the two libs and must export them in different vars
+            #  - NRNMECH_LIB_PATH the combined lib (used by neurodamus-py)
+            #  - BGLIBPY_MOD_LIBRARY_PATH is the pure mechanism (used by bglib-py)
+            if '_nd.' in libnrnmech_name:
+                run_env.set('NRNMECH_LIB_PATH', libnrnmech_name)
+            else:
+                run_env.set('BGLIBPY_MOD_LIBRARY_PATH', libnrnmech_name)
```

Comment:
can we avoid use of loop here?

Reply:
we are not sure about the order the libs are found